language_formatters_pre_commit_hooks/pretty_format_yaml.py

```diff
@@ -42,27 +65,38 @@ def pretty_format_yaml(argv=None):
     # Prevent ruamel.yaml to wrap yaml lines
     yaml.width = maxsize
 
+    separator = '---\n'
+
     for yaml_file in set(args.filenames):
         with open(yaml_file) as f:
             string_content = ''.join(f.readlines())
 
-        try:
-            content = yaml.load(string_content)
+        # Split multi-document file into individual documents
+        #
+        # Not using yaml.load_all() because it reformats primitive (non-YAML) content. It removes
+        # newline characters.
+        separator_pattern = '^---\\s*\\n'
```

Comment:
nit.
```suggestion
        separator_pattern = r'^---\s*\n'
```

Reply:
I thought I tried that originally and it didn't work. But it seems to be fine.